### DISCUSSION TITLE
Need a fork of otlpexporter in an internal copy of the collector

### DISCUSSION THREAD

**Discussion by peachchen0716** [open-telemetry/opentelemetry-collector-contrib#21485] [Q&A]
Hi, we would like to add some metrics on top of the existing otlpexporter in the core repository, and I would like to know what's the recommended way of forking a component from the core repo to our internal fork of the collector?

I can do straight copy/pasting, but would like to know if anything simpler? Thanks!

**Comment by atoulme** [comment]
That's a bit unclear to me. I would always start by engaging the community and opening an issue stating your intent and explaining which metrics you want to capture. Then I would offer changes in the form of a PR under the supervision of maintainers.

It's only if my changes don't get accepted that I would consider forking. Even while forking, I would keep working towards getting my changes incorporated to avoid running into maintenance and compatibility issues.

**Reply by peachchen0716** [reply]
Thanks for replying so quickly! The metrics I'm looking for is latency metric for the overall collector pipeline. It's essentially covered [here](https://github.com/open-telemetry/opentelemetry-collector/issues/542) as I looked around. I'm not sure if any recent change will address that, since [the last attempt](https://github.com/open-telemetry/opentelemetry-collector/pull/3183) seems to be closed due to [blocker here](https://github.com/open-telemetry/opentelemetry-collector/pull/3183#issuecomment-895040524).
